src/common.rs

```diff
@@ -1276,10 +1284,320 @@ fn path_directories() -> Vec<std::path::PathBuf> {
     dirs
 }
 
+pub fn get_delegated_validator_list_from_network(
+    network_config: &crate::config::NetworkConfig,
+) -> color_eyre::eyre::Result<Vec<near_primitives::types::AccountId>> {
+    Ok(get_validator_list(network_config)?
+        .into_iter()
+        .filter(|staking_pool_info| staking_pool_info.delegators.is_some())
+        .map(|staking_pool_info| staking_pool_info.validator_id)
+        .collect::<Vec<_>>())
+}
+
+fn get_used_delegated_validator_list_path(
+    credentials_home_dir: &std::path::Path,
+) -> std::path::PathBuf {
+    credentials_home_dir.join("delegated_validators.json")
+}
```

Comment:
We don't need to store separately previously selected staking pools. Let's discuss if offline